tests/io/test_api.py

```diff
@@ -31,15 +35,23 @@ def test_load_filter_nonchunked():
 
 @mark.api
 def test_load_monotonic():
-    data = aeon.load(monotonic_path, exp02.Patch2.Encoder)
+    data = aeon.load(monotonic_path, exp02.Patch2.Encoder, downsample=None)
     assert len(data) > 0 and data.index.is_monotonic_increasing
 
 
 @mark.api
 def test_load_nonmonotonic():
-    data = aeon.load(nonmonotonic_path, exp02.Patch2.Encoder)
+    data = aeon.load(nonmonotonic_path, exp02.Patch2.Encoder, downsample=None)
     assert not data.index.is_monotonic_increasing
 
 
+@mark.api
+def test_load_encoder_with_downsampling():
+    data = aeon.load(monotonic_path, exp02.Patch2.Encoder, downsample=True)
+    raw_data = aeon.load(monotonic_path, exp02.Patch2.Encoder, downsample=None)
+    assert len(data) < len(raw_data)
+    assert data.index.to_series().diff().dt.total_seconds().min() >= 0.02
```

Comment:
The point of downsampling is precisely to drop data points so that we have less data. It's not part of our data contract that we need to preserve the first and last points. The test seems pretty good to me.

Reply:
> However, it's important to note that while downsampling does indeed reduce the number of data points, it should not alter the overall time range of the data.

Preserving the "overall time range" of the data is fine, but this cannot be tested by naively comparing the first and last timestamps of the downsampled and raw data. When binning to a different frequency, e.g. 50 Hz we might need to introduce necessary rounding of bin timestamps to conform to the new regular sampling period. The "overall time range" will be correct, but the exact timestamps will be different, and in fact the test you suggested simply fails when run locally.